src/static_canvas.class.ts

```diff
@@ -35,7 +34,48 @@ import { CommonMethods } from './mixins/shared_methods.mixin';
    */
   // eslint-disable-next-line max-len
   fabric.StaticCanvas = fabric.util.createClass(
-    fabric.Collection,
+    class extends createCollectionMixin(CommonMethods) {
+      add(...objects: FabricObject[]) {
+        super.add(...objects);
+        objects.length > 0 && this.renderOnAddRemove && this.requestRenderAll();
```

Comment:
One day i would like to go over all of those `this.renderOnAddRemove` and decide if we really need them.
Those are good for demos because you have to write less, but sometimes boring during normal application development becuase sometimes you are doing something complex and you never want to render automatically.

Reply:
+1
Perhaps this is part of the scope of #8372